experiments/list_unsupported.sh

```diff
@@ -16,89 +16,10 @@ usage()
-  echo >&2 "  export PATH=\"${gnat2goto_bin}:\${PATH}\""
-  exit 1
-fi
-
-# gnat on the path?
-if ! command -v gnat > /dev/null; then
-   echo >&2 "Gnat not on PATH!"
-   # Take a wild guess at where gnat might be installed...
-   gnat_location="/opt/gnat/bin"
-   if [ -x "${gnat_location}/gnat" ] ; then
-      # Check if that's gnat 2016...
-      if gnat --version | grep -q 'GNAT GPL 2016' ; then
-         echo >&2 "Suggested adding gnat to your PATH with the following command:"
-         echo >&2 "  export PATH=\"${gnat_location}:\${PATH}\""
-      fi
-   fi
-   exit 4
-fi
-
-# Right version of gnat?
-if ! (gnat --version | grep -q 'GNAT GPL 2016') ; then
-   echo >&2 "The wrong version of gnat is on the PATH. gnat2goto requires GNAT GPL 2016."
-   exit 5
-fi
-
-# Sane ADA_INCLUDE_PATH and GPR_PROJECT_PATH?
-GNAT2GOTO=`command -v gnat2goto`
-ADA_HOME=`command -v gnat`
-ADA_HOME=$(cd "$(dirname ${ADA_HOME})/.." 2>/dev/null && pwd)
-PLATFORM=`${ADA_HOME}/bin/gcc -dumpmachine`
-DEF_ADA_INCLUDE_PATH="${ADA_HOME}/lib/gcc/${PLATFORM}"
-ADA_GCC_VERSION=`${ADA_HOME}/bin/gcc -dumpversion`
-DEF_ADA_INCLUDE_PATH="${DEF_ADA_INCLUDE_PATH}/${ADA_GCC_VERSION}/rts-native/adainclude:${ADA_HOME}/include"
-export ADA_INCLUDE_PATH="${ADA_INCLUDE_PATH:-$DEF_ADA_INCLUDE_PATH}"
-
-if [ -n "$GPR_PROJECT_PATH" -a ! -d "$GPR_PROJECT_PATH" ]; then
-   echo >&2 "GPR project path environment variable has been set to:"
-   echo >&2 "    \"${GPR_PROJECT_PATH}\""
-   echo >&2 "but that path is not a directory."
-   echo >&2 "Please set the environment variable GPR_PROJECT_PATH to the"
-   echo >&2 "location of the gnat libraries"
-   exit 6
-else
-   # GPR_PROJECT_PATH not previously specified, use a default
-   export GPR_PROJECT_PATH="${ADA_HOME}/lib/gnat"
-   if [ ! -d "$GPR_PROJECT_PATH" ]; then
-      echo >&2 "Could not find gnat library directory at ${GPR_PROJECT_PATH}"
-      echo >&2 "Please set the environment variable GPR_PROJECT_PATH to the"
-      echo >&2 "location of the gnat libraries"
-      exit 9
-   fi
-fi
-
-# Can we build the support tool?
-# Problem: GNAT 2016 helpfully installs a g++ binary alongside gnat... except
-# it's way out of date so invoking g++ via PATH looking will pickup that gnat g++
-# compiler, rather than the default system compiler... Instead we need to
-# temporarily drop the GNAT tools off the path while we build the tool.
-saved_path="$PATH"
-export PATH=$(echo ${PATH} | tr ':' '\n' | grep -v "${ADA_HOME}" | paste -s -d : - )
-experiment_dir=`dirname "$0"`
-gplusplus=`command -v g++`
-if ! ${gplusplus} --std=c++14 "${experiment_dir}/collect_unsupported.cpp" -o CollectUnsupported ; then
-   echo >&2 "Failed to compile support tool 'CollectUnsupported' using ${gplusplus}"
-   echo >&2 "You need a version of g++ on the PATH that supports C++14"
-   exit 8
-fi
-export PATH="${saved_path}"
-
-echo >&2 "...environment is OK."
+running_dir=$(dirname $(readlink -f $0))
```

Comment:
Should be surrounded with quotes (both commands), also @chrisr-diffblue correct me if I'm wrong but I believe `readlink -f` doesn't work on OSX?

Reply:
Correct - the `-f` means something different on OSX (and the `readlink`  tool is not in POSIX either). However, given its use here, I'm not quite sure why the readlink is even needed? I'd just use something like `. "$(dirname $0)/check_env_sanity.cfg"`

Reply:
@chrisr-diffblue the `readlink` is necessary to get the correct path if the tool is invoked via symlink. In this case this precaution is probably not necessary as it is unlikely to be exported to a global location.